zebra/dplane_fpm_nl.c

```diff
@@ -1264,6 +1326,7 @@ static int fpm_nl_process(struct zebra_dplane_provider *prov)
 	uint64_t cur_queue, peak_queue;
 
 	fnc = dplane_provider_get_data(prov);
+	frr_mutex_lock_autounlock(&fnc->ctxqueue_mutex);
```

Comment:
so this lock is moving  - and will be held throughout the loop below? that's quite a bit longer, and covers the 'provider_dequeue' api call, for example, which also has a lock. is there a specific situation you're trying to guard against with that change?

Reply:
I actually didn't hit any problems here, I just don't understand how this was working so I moved the `lock_autolock` outside of the loop to play safe and avoid double locking if it make any sense here.

Unless you think this is too big of critical area, then I guess I should move to manual lock/unlock so I can safely guard only the essential critical area inside the conditional it was before.

Reply:
I guess that's my concern - that the previous version used this mutex in a fairly specific, confined way, but in this version it's being held through more operations and api calls. unless there was a problem that needed to be fixed, I think I'd recommend the more limited scope?

Reply:
@eqvinox cleared this up: `frr_mutex_lock_autounlock` only lasts the scope block, so the previous code was fine. I fixed this by removing the commit.